File: src/me/pineman/arrays101/InsertingItems/DuplicateZeros.java
package me.pineman.arrays101.InsertingItems;

import java.util.Arrays;

public class DuplicateZeros {

    public static void main(String[] args) {
        int[] arr = {1,0,2,3,0,4,5,0};
        int[] result = duplicateZeros(arr);
        System.out.println(Arrays.toString(result));
    }

    public static int[] duplicateZeros(int[] arr) {
        int oglength = arr.length;
        boolean lastone = false;
        int i = 0;
        int remainder = 0;

        for (int num : arr) {
            if (num == 0) {
                if (lastone) {
                    lastone = false;
                }
                else {
                    for(int x=arr.length-1; x > i; x--){
                        arr[x] = arr[x-1];
                    }
                    arr[i] = 0;
                    lastone = true;
                }
            }
            else {
                lastone = false;
            }
            i++;
        }

        remainder = Math.abs(oglength - arr.length);

        for (int y = 0; y++ < remainder; y++) {
            Arrays.copyOf(arr, arr.length-1);
        }

        return arr;
    }
}

File: src/me/pineman/arrays101/InsertingItems/DuplicateZeros.py
arr = [1,0,2,3,0,4,5,0]

oglength = len(arr)
lastone = False
i = 0
remainder = 0

for num in arr:
    if num == 0:
        if lastone:
            lastone = False
        else:
            arr.insert(i, 0)
            lastone = True
    else:
        lastone = False
    i += 1

remainder = abs(oglength - len(arr))

for x in range(0, remainder):
    arr.pop()

print(arr)